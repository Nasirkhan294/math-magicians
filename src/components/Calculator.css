.calculator {
  width: 400px;
  background-color: #f2f2f2;
  border-radius: 5px;
}

.buttons-wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  border: 1px solid #858694;
}

.cal-display {
  background-color: #858694;
  padding: 20px;
  min-height: 60px;
  text-align: right;
  font-size: 20px;
  color: #fff;
}

.cal-digits {
  flex: 3;
  display: flex;
  flex-wrap: wrap;
}

.cal-operators {
  display: flex;
  flex-direction: column;
  flex: 1;
}

.cal-btn {
  flex: 1 1 33.3333%;
  min-height: 60px;
  border: none;
  border-right: 1px solid #c3c3c3;
  border-bottom: 1px solid #c3c3c3;
  outline: none;
  font-size: 16px;
}

.cal-btn:hover {
  background-color: #d3d3d4;
  color: #de6b0c;
}

.cal-operators .cal-btn {
  background-color: #f5913e;
  border-right: none;
}

.cal-operators .cal-btn:hover {
  background-color: #de6b0c;
  color: #fff;
}

.cal-digits .cal-btn:nth-of-type(13) {
  min-width: 199px;
}
