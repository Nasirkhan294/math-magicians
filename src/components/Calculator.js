import { useState } from 'react';
import DigitsButtons from './DigitsButtons';
import OperatorsButtons from './OperatorsButtons';
import './Calculator.css';
import calculate from './logic/calculate';

export default function Calculator() {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonName) => {
    const newData = calculate(state, buttonName);
    setState(newData);
  };

  return (
    <div className="calculator">
      <div className="cal-display">{state.next || state.total || '0'}</div>
      <div className="buttons-wrapper">
        <DigitsButtons handleButtonClick={handleButtonClick} />
        <OperatorsButtons handleButtonClick={handleButtonClick} />
      </div>
    </div>
  );
}
