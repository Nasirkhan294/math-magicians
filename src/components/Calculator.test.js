import { render } from '@testing-library/react';
import Calculator from './Calculator';

test('Navbar matches snapshot', () => {
  const { asFragment } = render(<Calculator />);
  expect(asFragment()).toMatchSnapshot();
});
