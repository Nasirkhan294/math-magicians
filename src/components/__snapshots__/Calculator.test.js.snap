// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="calculator-container"
  >
    <div
      class="cal-text"
    >
      <h3>
        Let's do some math!
      </h3>
    </div>
    <div
      class="calculator"
    >
      <div
        class="cal-display"
      >
        0
      </div>
      <div
        class="buttons-wrapper"
      >
        <div
          class="cal-digits"
        >
          <button
            class="cal-btn"
            type="button"
          >
            AC
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            +/-
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            %
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            7
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            8
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            9
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            4
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            5
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            6
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            1
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            2
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            3
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            0
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            .
          </button>
        </div>
        <div
          class="cal-operators"
        >
          <button
            class="cal-btn"
            type="button"
          >
            รท
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            x
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            -
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            +
          </button>
          <button
            class="cal-btn"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
